### PR TITLE
Feature : Added the functionality to identify user for each file being checked and updated functions for checking collab/non-collab files. (Pt-4)

### PR DESCRIPTION
**Pull request by 6s-1**
## 📝 Description

This PR is part-4 in a series of Prs for Non code file checker file and adds classification functionality and user detection to the capstone project. The implementation verifies user identity to ensure that the author being identified as the main contributor in our app is the same as the user using our app. Also, this PR identifies the files which the user has contributed to vs files other users have contributed to streamline downstream processes. Function to classify git repo detection has been updated to reuse existing code and local file detection is now integrated as well, Changes have been implemented according to offline team discussions as well.

**Key Features:**
- Git repository detection and metadata extraction
- verification of which files user contributed to vs files contributed to by other users. 
- User contribution verification in git repos
- User identity verification 
- Classification logic:
  - **Collaborative**: Git repo + 2+ authors + user is a contributor
  - **Non-collaborative**: Git repo with single author (user) OR local files (non-git)
- Reuses existing utilities from `git_utils.py` and `scan_utils.py` for consistency

**Closes:** #258 #259 #260

---

## 🔧 Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation added/updated
- [x] ✅ Test added/updated
- [x] ♻️ Refactoring
- [x] ⚡ Performance improvement

---

## 🧪 Testing

Comprehensive testing added for all functions mentioned with 100% tests passing. 

**To reproduce:**
```bash
# Install dependencies
pip install gitpython pytest pytest-mock

# Run tests
python3 -m pytest tests/utils/non_code_analysis/test_non_code_file_checker.py -v

# Run with coverage
python3 -m pytest tests/utils/non_code_analysis/test_non_code_file_checker.py -v --cov=app.utils.non_code_analysis.non_code_file_checker

```

## ✓ Checklist

- [x] 🤖 GenAI was used in generating the code and I have performed a self-review of my own code
- [x] 💬 I have commented my code where needed
- [x] 📖 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] ✅ I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [x] 🔗 Any dependent changes have been merged and published in downstream modules

---

### CODE REVIEW

**Review by KarimKhalil33 — CHANGES_REQUESTED** [review]
Great work shreya! just a few comments, 
- May we please remove the unused imports
- May we update the documentation regarding scan_project_files since thats not being used anymore here and the part mentioning: Ensures collaborative files have user + at least 1 other person. Ensures non-collaborative files have only the user. But the function doesn’t actually decide or validate collaboration status, it just groups files based on the authors list.
- also, im not sure if we are using pytest-mock because i only see unittest.mock (Mock, MagicMock, patch) if not, can we also remove that from req.txt and the import call

Aside from these minor comments, the code looks great! thank you

**Review by KarimKhalil33 — APPROVED** [review]
Thanks for addressing the comments!

File: app/utils/non_code_analysis/non_code_file_checker.py [hunk 1]
```diff
@@ -142,3 +181,75 @@ def filter_non_code_files_by_collaboration(
         "collaborative": collaborative,
         "non_collaborative": non_collaborative
     }
+
+def get_git_user_identity(repo_path: Union[str, Path]) -> Dict[str, str]:
```
Comment (dabby04) [inline app/utils/non_code_analysis/non_code_file_checker.py]: Great work for this method! I think this should be moved into `git_utils.py` since this handles mostly `.git` logic

File: tests/utils/non_code_analysis/test_non_code_file_checker.py [hunk 2]
```diff
@@ -965,3 +988,110 @@ def test_filter_by_collaboration_parametrized(author_count, threshold, expected_
...
+def test_get_git_user_identity_invalid_repo(mock_get_repo):
+    """Test handling invalid repository."""
+    mock_get_repo.side_effect = Exception("Not a git repo")
+    result = get_git_user_identity('/invalid/path')
+    assert result == {}
+
+
+
+# ============================================================================
+# Tests for verify_user_in_files()
```
Comment (PaintedW0lf) [inline tests/utils/non_code_analysis/test_non_code_file_checker.py]: Great work. Do you think we would need these extra lines for overall comments when we already have the definitions underneath?

**Review by PaintedW0lf — DISMISSED** [review]
Over all great work done, and the test coverage is impressively thorough across all new branches and edge cases.
Few things i added in comments underneath with some suggestion - 

1) is_directory_git_repo() might be unnecessary duplication
Since the method just wraps detect_git(), consider whether it’s needed at all unless there’s a planned abstraction. Calling detect_git() directly may make the flow simpler and avoid an extra layer.
2) also there are repeated header sections (# ==========================================================================). With the docstrings already present, if you think its better we can consider reducing some of these for readability.

Thanks for your work again :) !!

File: tests/utils/non_code_analysis/test_non_code_file_checker.py (hunk 2 above)
Comment (6s-1) [reply tests/utils/non_code_analysis/test_non_code_file_checker.py]: I can take care of that in upcoming PRs, thanks!

File: app/utils/non_code_analysis/non_code_file_checker.py (hunk 1 above)
Comment (6s-1) [reply app/utils/non_code_analysis/non_code_file_checker.py]: Left reasoning in comments.

**Comment by 6s-1** [comment]
> Over all great work done, and the test coverage is impressively thorough across all new branches and edge cases. Few things i added in comments underneath with some suggestion -
> 
> 1. is_directory_git_repo() might be unnecessary duplication
>    Since the method just wraps detect_git(), consider whether it’s needed at all unless there’s a planned abstraction. Calling detect_git() directly may make the flow simpler and avoid an extra layer.
> 2. also there are repeated header sections (# ==========================================================================). With the docstrings already present, if you think its better we can consider reducing some of these for readability.
> 
> Thanks for your work again :) !!

Thanks for your feedback!

1. Was already brought up in other comments and resolved.
2. There are no repeated header sections, all are unique. It is for easier readability for other developers working on the files as the test file has 90 + test cases so it helps to have separate section headers for checkpoints. Also each function has ~10-15 tests. The headers group related tests together.

Thank you for your feedback! Unclear why this is marked as requested changes as there are no clear changes other than ones already requested by other reviewers, hence resolving the requested changes here.

**Review by dabby04 — APPROVED** [review]
Thank you for your explanation on why you are keeping the method there. I understand this, everything else looks good to me.
